routes/movies.routes.js

```diff
@@ -0,0 +1,67 @@
+const router = require("express").Router();
+const movieModel = require("../models/Movie.model")
+const celebrityModel = require("../models/Celebrity.model");
+const MovieModel = require("../models/Movie.model");
+
+
+router.get("/movies", (req, res, next) => {
+    
+    movieModel.find()
+    .then((movies) => {
+        res.render("movies/movies", {moviesArr: movies})
+    })
+    .catch( e => {
+        console.log("error creating new movie", e);
+        next(e);
+    });
+})
+
+router.get("/movies/create", (req, res, next) => {
+    celebrityModel.find()
+        .then((celebritiesFromDB) => {
+        res.render("movies/new-movie", {celebritiesArr: celebritiesFromDB})
+    })
+    .catch( e => {
+        console.log("error creating new movie", e);
+        next(e);
+    });
+})
+
+router.post("/movies/create", (req, res, next) => {
+    
+    const newMovie = {
+        title: req.body.title,
+        genre: req.body.genre,
+        plot: req.body.plot,
+        cast: req.body.cast
+    };
+    MovieModel.create(newMovie)
+    .then( (newMovie) => {
+        res.redirect("/movies");
+    })
+    .catch( e => {
+        console.log("error creating new movie", e);
+        next(e);
+    });
+});
+
+router.get("/movies/:id", (req, res, next) => {
+    const id = req.params.id;
+
+    movieModel.findById(id)
+    .populate("cast")
+    .then((moviesFromDB) => {
+        const dataMovie = {
+            moviesArr: moviesFromDB
+        }
+        console.log(moviesArr, _id)
+        res.render("movies/movie-details", dataMovie)
```

Comment:
`.findById()` gets just one document, so you could have passed that directly to the view

```suggestion
        res.render("movies/movie-details", moviesFromDB )
```